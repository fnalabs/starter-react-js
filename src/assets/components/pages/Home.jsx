import React, { Component, Fragment } from 'react'
import Helmet from 'react-helmet'
import { Cookies } from 'react-cookie-consent'
import ReactGA from 'react-ga'

import meta from 'metadata.json'

export default class Home extends Component {
  componentDidMount () {
    if (Cookies.get('CookieConsent')) {
      ReactGA.pageview(this.props.location.pathname, undefined, meta.common.siteName)
    }
  }

  render () {
    const { description, url } = meta['/']
    const siteName = meta.common.siteName

    return (
      <Fragment>
        <Helmet>
          <title>{siteName}</title>
          <meta name='description' content={description} />
          <link rel='canonical' href={url} />

          <meta property='og:title' content={siteName} />
          <meta property='og:description' content={description} />
          <meta property='og:site_name' content={siteName} />
          <meta property='og:url' content={url} />
          <meta property='og:type' content='website' />
        </Helmet>

        <div>Homepage content</div>
      </Fragment>
    )
  }
}
