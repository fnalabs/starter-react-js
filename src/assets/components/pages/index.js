// site's main pages
export { default as Home } from './Home.jsx'

// policy pages
export { Cookie } from './policies/Cookie.jsx'
export { Privacy } from './policies/Privacy.jsx'
