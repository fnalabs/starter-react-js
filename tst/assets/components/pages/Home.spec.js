/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import { Cookies } from 'react-cookie-consent'
import ReactGA from 'react-ga'

import { Home } from 'pages'

jest.mock('react-ga')

describe('<Home />', () => {
  test('should render homepage static content', () => {
    Cookies.get = jest.fn().mockReturnValue(false)

    const tree = renderer.create(<Home location={{ pathname: '/' }} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(ReactGA.pageview).not.toBeCalled()
  })

  test('should render homepage static content and report page view', () => {
    Cookies.get = jest.fn().mockReturnValue(true)

    const tree = renderer.create(<Home location={{ pathname: '/' }} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(ReactGA.pageview).toBeCalledTimes(1)
  })
})
