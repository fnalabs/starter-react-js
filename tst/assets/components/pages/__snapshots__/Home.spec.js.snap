// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render homepage static content 1`] = `
<div>
  Homepage content
</div>
`;
